tests/unit/test-extension-runners/test.chromium.js

```diff
@@ -357,4 +368,71 @@ describe('util/extension-runners/chromium', async () => {
     await runnerInstance.exit();
   });
 
+  describe('reloadAllExtensions', () => {
+    let runnerInstance;
+    let wsClient;
+
+    beforeEach(async () => {
+      const {params} = prepareExtensionRunnerParams();
+      runnerInstance = new ChromiumExtensionRunner(params);
+      await runnerInstance.run();
+    });
+
+    const connectClient = async () => {
+      const wssInfo = runnerInstance.wss.address();
+      const wsURL = `ws://${wssInfo.address}:${wssInfo.port}`;
+      wsClient = new WebSocket(wsURL);
+      await new Promise((resolve) => wsClient.on('open', resolve));
+    };
+
+    afterEach(async () => {
+      wsClient.close();
+      await runnerInstance.exit();
+    });
+
+    it('does not resolve before complete message from client', async (done) => {
```

Comment:
I modified the test based on https://github.com/mozilla/web-ext/pull/1695#discussion_r324148564, but couldn't make this pass yet.

I'm setting a variable (`reloadMessage`), creating a client and registering a message listener inside a promise (`waitCompleteMessage`), that change the variable value and resolve when receive complete message. Add a timeout and await `reloadAllExtensions`.

The timeout is asserting that variable isn't change, send complete message, await `waitCompleteMessage` and assert variable was changed.

But `waitCompleteMessage` isn't resolving. I was adding some logs and notice that the listener doesn't called after sent complete message, even when is added before `reloadAllExtensions`.

Do you have any advice?

Reply:
I finally fix this test. I'm using just one flag that check is false before and become true after `reloadAllExtensions`, but let me know if you prefer to use two flags instead.